### PR TITLE
tests for drivers endpoints

### PR DESCRIPTION
**Pull request by harrisonchin2523**
### Summary 
 These are the tests for the drivers endpoints listed in the Notion.

https://dti-carriage.notion.site/Carriage-API-Doc-e78892c2e23e42259aa332b5b3136f22#d7fecf7c4dfd4985823b517bb75b24bc
### Test Plan 

Run "npm run test"

### CODE REVIEW

File: server/tests/driver.test.ts [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+  // testing the addition of a new driver
+
+  describe('Create a new driver', () => {
+    const generateCreateDriverTest = async (authToken: string) => {
+      const newDriverData = {
+        email: 'drivertest3-email@test.com',
+        phoneNumber: '1234567890',
+        firstName: 'Test',
+        lastName: 'Testing',
+        vehicle: testVehicles[2],
```
Comment (pratyush1712) [inline server/tests/driver.test.ts:253]: use vehicle id

**Review by pratyush1712 — DISMISSED** [review]
please write tests for stats endpoint and use vehicle's id everywhere. looks good, otherwise!

File: server/tests/driver.test.ts [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+  },
+];
+const testDrivers: DriverType[] = [
+  {
+    id: 'driver0',
+    email: 'drivertest-email@test.com',
+    phoneNumber: '1234567890',
+    firstName: 'Test',
+    lastName: 'Testing',
+    vehicle: testVehicles[0],
```
[the remark this answers is not in the document]
Comment (harrisonchin2523) [reply server/tests/driver.test.ts:60]: The problem with using vehicle ID is that when I retrieve a rider, the entire vehicle object is stored in the vehicle field, not the ID which causes the test to fail, does it have to be the vehicle ID? If it does, then I can definitely find a workaround to change this.

[the remark this answers is not in the document]
Comment (pratyush1712) [reply server/tests/driver.test.ts:60]: i think it's supposed to work that way only. So, you make a POST request with vehicle id, and when you do a GET request it should return the vehicle object (not the id). Database stores vehicle id bu they get request converts the vehicle id to vehicle object.
`POST -> {vehicle: "id",...}` and `GET -> retrieves {vehicle: "id", ...} -> converts to {vehicle:{name:"ABC"}}`

[the remark this answers is not in the document]
Comment (harrisonchin2523) [reply server/tests/driver.test.ts]: Gotcha! The most recent commit should address this.

**Comment by harrisonchin2523** [comment]
When trying to populate the mock database with a new ride, I noticed that none of these rides were actually popping up at all. This seems to be an issue with the RideType, specifically with startLocation and endLocation. Theoretically, we should be providing a location ID to do so because the database should populate the data with the use of an ID, but instead, it requires an entire RideLocation object. I'm not too sure how to progress with this issue because when trying to change the type in RideType to RideLocation | string, errors come up across the rest of the codebase in the router file for rides (rides.ts) and in notificationMsg.ts in utils.

**Review by pratyush1712 — COMMENTED** [review]
Looks good to me! Please implement the notifications test and it should be good to merge

**Review by pratyush1712 — APPROVED** [review]
lgtm!

**Review by pratyush1712 — APPROVED** [review]
lgtm! thanks for implementing driver tests

**Pull request merged**